data_safe_haven/administration/users/user_handler.py

```diff
@@ -163,36 +156,60 @@ def set(self, users_csv_path: str) -> None:
             DataSafeHavenUserHandlingException if the users could not be set to the desired list
         """
         try:
-            # # Construct user lists
-            # with open(users_csv_path, "r", encoding="utf-8") as f_csv:
-            #     new_users = [
-            #         ResearchUser.from_csv(
-            #             account_enabled=True,
-            #             domain_suffix=self.cfg.azure.domain_suffix,
-            #             **user,
-            #         )
-            #         for user in csv.DictReader(f_csv, delimiter=";")
-            #     ]
-            # for new_user in new_users:
-            #     self.debug(f"Processing user: {new_user}")
-
-            # # Keep existing users with the same username
-            # azuread_users = [user for user in self.azuread.users if user in new_users]
-            # guacamole_users = [
-            #     user for user in self.guacamole.users if user in new_users
-            # ]
-
-            # # Add any new users
-            # azuread_users += [user for user in new_users if user not in azuread_users]
-            # guacamole_users += [
-            #     user for user in new_users if user not in guacamole_users
-            # ]
-
-            # # Commit changes
-            # self.azuread.set(azuread_users)
-            # self.guacamole.set(guacamole_users)
-            pass
+            # Construct user list
+            with open(users_csv_path, encoding="utf-8") as f_csv:
+                reader = csv.DictReader(f_csv)
+                for required_field in ["GivenName", "Surname", "Phone", "Email"]:
+                    if (not reader.fieldnames) or (
+                        required_field not in reader.fieldnames
+                    ):
+                        raise ValueError(
+                            f"Missing required CSV field '{required_field}'."
+                        )
+                desired_users = [
+                    ResearchUser(
+                        country="GB",
```

Comment:
Should this be hardcoded?

Is this a limitation on phone numbers we support?

Reply:
Fixed